src/OSPSuite.Presentation/Presenters/Charts/DataBrowserPresenter.cs

```diff
@@ -155,6 +168,29 @@ public void UsedChangedFor(DataColumnDTO dataColumnDTO, bool used)
       {
          raiseUsedChanged(dataColumnDTO.DataColumn, used);
          updateDataSelection(_view.SelectedColumns);
+         updateLinkedObservedData(dataColumnDTO.DataColumn, used);
+      }
+
+      private void updateLinkedObservedData(DataColumn dataColumn, bool used)
+      {
+         if (!_isLinkedMappedOutputs) return;
+         
+         var linkedObservedData = getLinkedObservedDataFromOutputPath(dataColumn.PathAsString);
+         SetUsedState(linkedObservedData, used);
+      }
+
+      private List<DataColumnDTO> getLinkedObservedDataFromOutputPath(string updatedDataColumnName)
+      {
+         var linkedObservedDataRepositories = AllOutputMappings.AllDataRepositoryMappedTo(updatedDataColumnName);
+         var linkedObservedData = getDataColumnDTOsFromDatarepositories(linkedObservedDataRepositories);
+         return linkedObservedData;
+      }
+
+      private List<DataColumnDTO> getDataColumnDTOsFromDatarepositories(IEnumerable<DataRepository> linkedObservedDataRepositories)
+      {
+         var linkedObservedData = _dataColumnDTOCache.KeyValues.Where(x => linkedObservedDataRepositories.Any(y => y.Id == x.Key.Repository.Id))
```

Comment:
Make the code easier to read. Where, Any, Select
This can be simplified by not using the primitive  such as Id but working on lists directly
Make the parameter IReadOnly
IReadOnlyList<DataRepository> linkedObservedDataRepositories

and then 
return _dataColumnDTOCache.Where(x => linkedObservedDataRepositories.Contains(x.DataColumn.Repository)).ToList();

you can also create a Repistory method in the DataColumn DTO Repository =>DataColum?.Repository
so it becomes
return _dataColumnDTOCache.Where(x => linkedObservedDataRepositories.Contains(x.Repository)).ToList();

In general, there is no reason to work with dictionary, cache, key and values at the presenter level. it is possible, but often not required

Reply:
used the IReadOnlyList  and direct return. I would say creating a new Repository method in DataColumnDTO is an overkill: to me it does not simplify the code that much